### PR TITLE
feat(vtkGCodeReader): add vtkGCodeReader

### LINKED ISSUE #3149: [Feature] Support for G-Code format
### Motivation

Support for the G-Code file format

### Detailed Description

Add a new vtkGCodeReader

```js
import vtkGCodeReader from '@kitware/vtk.js/IO/Misc/GCodeReader';

const reader = vtkGCodeReader.newInstance();
reader.parse(fileReader.result);
const numLayers = reader.getNumberOfOutputPorts();
for (let idx = 1; idx < numLayers; idx++) {
    const polyData = reader.getOutputData(idx);
    [...]
}
```

Spec https://linuxcnc.org/docs/html/gcode/overview.html

### PR DESCRIPTION
**Pull request by daker**
### Context
This PR adds support for G-Code format
Fixes #3149

### Results

https://github.com/user-attachments/assets/e5531269-315d-40dc-bf3d-335ef94a3175

### Changes
- [x] Documentation and TypeScript definitions were updated to match those changes

### PR and Code Checklist

- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Testing

- [ ] This change adds or fixes unit tests 
- [x] Tested environment:
  - **vtk.js**: 
  - **OS**: 
  - **Browser**:

### CODE REVIEW

File: Sources/IO/Misc/GCodeReader/example/index.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+let renderer = null;
+let renderWindow = null;
+let layersSelector = null;
+let layersLabel = null;
+
+const reader = vtkGCodeReader.newInstance();
+
+// ----------------------------------------------------------------------------
+
+function hslToRgb(h, s, l) {
```
Comment (jourdain) [inline Sources/IO/Misc/GCodeReader/example/index.js:26]: Don't we have something like that in vtkMath?

Comment (daker) [reply Sources/IO/Misc/GCodeReader/example/index.js]: Yes, fixed the example

File: Sources/IO/Misc/GCodeReader/index.js [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+import vtkPoints from 'vtk.js/Sources/Common/Core/Points';
+import vtkCellArray from 'vtk.js/Sources/Common/Core/CellArray';
+
+// Enable several sources for DataAccessHelper
+import 'vtk.js/Sources/IO/Core/DataAccessHelper/LiteHttpDataAccessHelper'; // Just need HTTP
+// import 'vtk.js/Sources/IO/Core/DataAccessHelper/HttpDataAccessHelper'; // HTTP + gz
+// import 'vtk.js/Sources/IO/Core/DataAccessHelper/HtmlDataAccessHelper'; // html + base64 + zip
+// import 'vtk.js/Sources/IO/Core/DataAccessHelper/JSZipDataAccessHelper'; // zip
+
+const state = {};
```
Comment (jourdain) [inline Sources/IO/Misc/GCodeReader/index.js:14]: Is it supposed to be global?

File: Sources/IO/Misc/GCodeReader/index.js [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+// ----------------------------------------------------------------------------
+function vtkGCodeReader(publicAPI, model) {
+  state.currentPosition = { x: 0, y: 0, z: 0 };
+  state.offset = { x: 0, y: 0, z: 0 };
+  state.currentLayer = 0;
+  state.layers = new Map(); // Map to store layer data
+  state.isAbsolute = true; // G90 is default
+  state.isMetric = true; // G21 is default
+  state.lastZ = 0; // Track Z changes for layer detection
+
```
Comment (jourdain) [inline Sources/IO/Misc/GCodeReader/index.js:27]: Any new instance will reset the global state.

File: Sources/IO/Misc/GCodeReader/index.js (hunk 2 above)
Comment (daker) [reply Sources/IO/Misc/GCodeReader/index.js:14]: no it's related to the reader, once the parser is done it will be not used

Comment (daker) [reply Sources/IO/Misc/GCodeReader/index.js:14]: Maybe i should declare it inside vtkGCodeReader function

Comment (jourdain) [reply Sources/IO/Misc/GCodeReader/index.js]: yes you can define it inside the closure scope.

Comment (daker) [reply Sources/IO/Misc/GCodeReader/index.js]: Yes that's what i did

**Review by jourdain — APPROVED** [review]

**Comment by finetjul** [comment]
Instead of having multiple polydatas as output of your reader, you should have a unique polydatas output, but for each cell, you associate a layer scalar value into a cell data array. 
For rendering, you can filter the polydatas based on the cell data array. For coloring you apply a unique look up table with a color per layer value.

**Pull request merged**